theano/tensor/basic.py

```diff
@@ -276,7 +276,9 @@ def constant(x, name=None, ndim=None, dtype=None):
     if (sig not in constant_cache and ret.data.size == 1 and
         (-10) <= ret.data <= 10 and
         (ret.dtype in int_dtypes or ret.dtype in uint_dtypes or
-         (ret.dtype in float_dtypes and int(ret.data) == ret.data))):
+         (ret.dtype in float_dtypes and
+          # Limit the size of the cache.
+          len(constant_cache) < 10000))):
```

Comment:
I think a hard cut-off is better then the current state of no limit. Especially as the cut-off is high.

We can't make an ordinary LRU or LFU easily. We can only remove constant that aren't in the cache anymore. But I think this is overkill. This is just a cache for constant. There isn't good reason to have more then 10000 values. If that happen, then the user have very big graph and have others problems too.